lib/workers/repository/update/pr/pr-cache.ts

```diff
@@ -0,0 +1,43 @@
+import { logger } from '../../../../logger';
+import { getCache } from '../../../../util/cache/repository';
+import type { PrCache } from '../../../../util/cache/repository/types';
+
+export function getPrCache(branchName: string): PrCache | null {
+  logger.debug(`getPrCache()`);
+  const cache = getCache();
+  const branch = cache.branches?.find(
+    (branch) => branchName === branch.branchName
+  );
+
+  if (branch?.prCache) {
+    return branch.prCache;
+  }
+
+  return null;
+}
+
+// store the time a PR was last updated
+export function setPrCache(
+  branchName: string,
+  fingerprint: string,
+  prModified: boolean
+): void {
+  logger.debug(`setPrCache()`);
+  const cache = getCache();
+  const branch = cache.branches?.find(
+    (branch) => branchName === branch.branchName
+  );
+
+  if (!branch) {
+    logger.debug(`setPrCache(): Branch cache not present`);
```

Comment:
Shouldn't there be a cache created if there is non?

Reply:
If there is no branchCache we don't create a new one, it was decided during the redoing of branchCache logic a while back. Will link the issue/comment once I find it.